src/main/scala/viper/server/vsi/HTTP.scala

```diff
@@ -68,15 +68,20 @@ trait VerificationServerHttp extends VerificationServer with CustomizableHttp {
   private var stoppedPromise: Promise[Unit] = _
   var bindingFuture: Future[Http.ServerBinding] = _
 
-  /** The server has been stopped when `stopped` completes. Afterwards, the execution context can be terminated. */
-  override def start(active_jobs: Int): Unit = {
+  /** The server has been stopped when `stopped` completes. Afterwards, the execution context can be terminated.
+    * The returned future completes as soon as the server has been started. Before completion, `port` will be set
+    * to the port number selected by the HTTP server
+    */
+  override def start(active_jobs: Int): Future[Unit] = {
     ast_jobs = new JobPool("AST-pool", active_jobs)
     ver_jobs = new JobPool("Verification-pool", active_jobs)
     stoppedPromise = Promise()
     bindingFuture = Http().newServerAt("localhost", port).bindFlow(setRoutes())
-    bindingFuture.map(_.whenTerminationSignalIssued)
     _termActor = system.actorOf(Terminator.props(ast_jobs, ver_jobs, Some(bindingFuture)), "terminator")
-    isRunning = true
+    bindingFuture.map { serverBinding =>
+      port = serverBinding.localAddress.getPort
```

Comment:
Do I understand correctly that the port is reassigned iff it has been initially set to 0? If yes, please add this in a comment.

Reply:
It is reassigned in all cases. However, the port number should not change when a non-zero port number was initially used. I'll add a corresponding assertion